test/framework/toolchain.py

```diff
@@ -189,7 +169,7 @@ def test_validate_pass_by_value(self):
     def test_optimization_flags(self):
         """Test whether optimization flags are being set correctly."""
 
-        flag_vars = ['CFLAGS', 'CXXFLAGS', 'FFLAGS', 'F90FLAGS']
+        flag_vars = ['CFLAGS', 'CXXFLAGS', 'FCFLAGS', 'FFLAGS', 'F90FLAGS']
```

Comment:
It's only a test so it doesn't really matter, but at basically all other places you used the order F77, F90, FC. Maybe adopt it here (and 6 more times below) as well?